src/sentry/static/sentry/app/components/group/externalIssueActions.tsx

```diff
@@ -1,207 +1,120 @@
-    this.props.onChange(() => onSuccess());
-    this.closeModal();
   };
 
-  renderBody() {
-    const {action, selectedIntegration} = this.state;
-    const {linked, unlinked} = this.linkedIssuesFilter();
-    return (
-      <React.Fragment>
-        {linked.length > 0 &&
-          linked.map(config => {
-            const {provider, externalIssues} = config;
-            const issue = externalIssues[0];
-            return (
-              <IssueSyncListElement
-                key={issue.id}
-                externalIssueLink={issue.url}
-                externalIssueId={issue.id}
-                externalIssueKey={issue.key}
-                externalIssueDisplayName={issue.displayName}
-                onClose={() => this.deleteIssue(config)}
-                integrationType={provider.key}
-                hoverCardHeader={t('Linked %s Integration', provider.name)}
-                hoverCardBody={
-                  <div>
-                    <IssueTitle>{issue.title}</IssueTitle>
-                    {issue.description && (
-                      <IssueDescription>{issue.description}</IssueDescription>
-                    )}
-                  </div>
-                }
-              />
-            );
-          })}
-
-        {unlinked.length > 0 && (
+  const doOpenModal = (integration: GroupIntegration) =>
+    openModal(deps => (
+      <ExternalIssueModal {...deps} {...{configurations, group, onChange, integration}} />
+    ));
+
+  return (
+    <React.Fragment>
+      {linked.map(config => {
+        const {provider, externalIssues} = config;
+        const issue = externalIssues[0];
+        return (
           <IssueSyncListElement
-            integrationType={unlinked[0].provider.key}
-            hoverCardHeader={t('Linked %s Integration', unlinked[0].provider.name)}
+            key={issue.id}
+            externalIssueLink={issue.url}
+            externalIssueId={issue.id}
+            externalIssueKey={issue.key}
+            externalIssueDisplayName={issue.displayName}
+            onClose={() => deleteIssue(config)}
+            integrationType={provider.key}
+            hoverCardHeader={t('Linked %s Integration', provider.name)}
             hoverCardBody={
-              <Container>
-                {unlinked.map(config => (
-                  <Wrapper onClick={() => this.openModal(config)} key={config.id}>
-                    <IntegrationItem integration={config} />
-                  </Wrapper>
-                ))}
-              </Container>
+              <div>
+                <IssueTitle>{issue.title}</IssueTitle>
+                {issue.description && (
+                  <IssueDescription>{issue.description}</IssueDescription>
+                )}
+              </div>
             }
-            onOpen={unlinked.length === 1 ? () => this.openModal(unlinked[0]) : undefined}
-            showHoverCard={this.state.showModal ? false : undefined}
           />
-        )}
-        {selectedIntegration && (
```

Comment:
@EvanPurkhiser do we not need a check to see if the `selectedIntegration` exists anymore?

Reply:
No since once the modal is open it's selected. This was only really helpful when we kept modal state in this, now that state is kept as a prop to the modal